planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/utils.cpp

```diff
@@ -182,21 +183,91 @@ bool path_footprint_exceeds_target_lane_bound(
+  for (const auto & p : target_lane_reference_path.points) {
+    xs.push_back(p.point.pose.position.x);
+    ys.push_back(p.point.pose.position.y);
+  }
+  sampler_common::transform::Spline2D reference_spline(xs, ys);
+  frenet_planner::FrenetState initial_state;
+  const auto & initial_position = lane_change_info.lane_changing_start.position;
+  initial_state.position = reference_spline.frenet({initial_position.x, initial_position.y});
+  initial_state.longitudinal_velocity = lane_change_velocity.prepare;
+  initial_state.longitudinal_acceleration = longitudinal_acceleration.prepare;
+  const auto initial_yaw = tf2::getYaw(lane_change_info.lane_changing_start.orientation);
+  const auto initial_s = initial_state.position.s + 0.001;  // add epsilon to avoid s=0
+  initial_state.lateral_velocity = 0.0;
+  initial_state.lateral_acceleration = 0.0;
+  frenet_planner::SamplingParameters sampling_parameters;
+  sampling_parameters.parameters.emplace_back();
+  sampling_parameters.resolution = 0.2;
+  sampling_parameters.parameters.back().target_duration = lane_change_info.duration.lane_changing;
+  auto & target_state = sampling_parameters.parameters.back().target_state;
+  target_state.position = reference_spline.frenet(
+    {lane_change_info.lane_changing_end.position.x, lane_change_info.lane_changing_end.position.y});
+  target_state.position.d = 0.0;
+  target_state.longitudinal_velocity = lane_change_velocity.lane_changing;
+  target_state.longitudinal_acceleration = 0.0;  // it is OK to not use longitudinal_acceleration
+  // target lateral velocity is based on the initial lateral velocity relative to the target point
+  // TODO(Maxime): not sure if we should use curvature at initial or target s
+  target_state.lateral_velocity =
+    (1 - reference_spline.curvature(initial_s) * initial_state.position.d) *
+    std::tan(initial_yaw - reference_spline.yaw(target_state.position.s));
+  target_state.lateral_acceleration = 0.0;
+  const auto candidates =
+    frenet_planner::generateTrajectories(reference_spline, initial_state, sampling_parameters);
+  const auto & candidate = candidates.front();
+  PathPointWithLaneId pp;
+  auto ref_s = 0.0;
+  auto ref_i = 0UL;
+  for (auto i = 0UL; i < candidate.poses.size(); ++i) {
+    const auto s = candidate.frenet_points[i].s;
+    pp.point.pose = candidate.poses[i];
+    pp.point.longitudinal_velocity_mps = static_cast<float>(candidate.longitudinal_velocities[i]);
+    pp.point.lateral_velocity_mps = static_cast<float>(candidate.lateral_velocities[i]);
+    // copy from original reference path
+    while (ref_s < s && ref_i + 1 < target_lane_reference_path.points.size()) {
+      ++ref_i;
+      ref_s += universe_utils::calcDistance2d(
+        target_lane_reference_path.points[ref_i - 1], target_lane_reference_path.points[ref_i]);
+    }
+    pp.point.pose.position.z = target_lane_reference_path.points[ref_i].point.pose.position.z;
+    pp.lane_ids = target_lane_reference_path.points[ref_i].lane_ids;
```

Comment:
```suggestion
  ShiftedPath shifted_path;
  std::vector<double> xs;
  std::vector<double> ys;
  xs.reserve(target_lane_reference_path.points.size());
  ys.reserve(target_lane_reference_path.points.size());
  for (const auto & p : target_lane_reference_path.points) {
    xs.push_back(p.point.pose.position.x);
    ys.push_back(p.point.pose.position.y);
  }
  sampler_common::transform::Spline2D reference_spline(xs, ys);
  
  frenet_planner::FrenetState initial_state;
  const auto & initial_position = lane_change_info.lane_changing_start.position;
  initial_state.position = reference_spline.frenet({initial_position.x, initial_position.y});
  initial_state.longitudinal_velocity = lane_change_velocity.prepare;
  initial_state.longitudinal_acceleration = longitudinal_acceleration.prepare;
  const auto initial_yaw = tf2::getYaw(lane_change_info.lane_changing_start.orientation);
  const auto initial_s = initial_state.position.s + 0.001;  // add epsilon to avoid s=0
  initial_state.lateral_velocity = 0.0;
  initial_state.lateral_acceleration = 0.0;
  
  frenet_planner::SamplingParameters sampling_parameters;
  sampling_parameters.parameters.emplace_back();
  sampling_parameters.resolution = 0.2;
  sampling_parameters.parameters.back().target_duration = lane_change_info.duration.lane_changing;
  
  auto & target_state = sampling_parameters.parameters.back().target_state;
  target_state.position = reference_spline.frenet(
    {lane_change_info.lane_changing_end.position.x, lane_change_info.lane_changing_end.position.y});
  target_state.position.d = 0.0;
  target_state.longitudinal_velocity = lane_change_velocity.lane_changing;
  target_state.longitudinal_acceleration = 0.0;  // it is OK to not use longitudinal_acceleration
  // target lateral velocity is based on the initial lateral velocity relative to the target point
  // TODO(Maxime): not sure if we should use curvature at initial or target s
  target_state.lateral_velocity =
    (1 - reference_spline.curvature(initial_s) * initial_state.position.d) *
    std::tan(initial_yaw - reference_spline.yaw(target_state.position.s));
  target_state.lateral_acceleration = 0.0;
  
  const auto candidates =
    frenet_planner::generateTrajectories(reference_spline, initial_state, sampling_parameters);
  const auto & candidate = candidates.front();
  
  PathPointWithLaneId pp;
  auto ref_s = 0.0;
  auto ref_i = 0UL;
  for (auto i = 0UL; i < candidate.poses.size(); ++i) {
    const auto s = candidate.frenet_points[i].s;
    pp.point.pose = candidate.poses[i];
    pp.point.longitudinal_velocity_mps = static_cast<float>(candidate.longitudinal_velocities[i]);
    pp.point.lateral_velocity_mps = static_cast<float>(candidate.lateral_velocities[i]);
    // copy from original reference path
    while (ref_s < s && ref_i + 1 < target_lane_reference_path.points.size()) {
      ++ref_i;
      ref_s += universe_utils::calcDistance2d(
        target_lane_reference_path.points[ref_i - 1], target_lane_reference_path.points[ref_i]);
    }
    pp.point.pose.position.z = target_lane_reference_path.points[ref_i].point.pose.position.z;
    pp.lane_ids = target_lane_reference_path.points[ref_i].lane_ids;
```

Reply:
Sorry it was hurting my eye